### PR TITLE
Check if filename is a URL before checking if file exists

### PR DESCRIPTION
**Pull request by tiangolo**
Check if filename is a URL before checking if file exists.

It does so by parsing the `filename` and checking if it's a URL with a scheme (the "http" or "rstp" in "http://youtube.com/..." or in "rstp://192.168.1.15:5544/") before parsing if a file exists. So, if the path is a file, it will still do the check.

This should solve #35.

I needed it so I added it (and I tested it, I'm already using this fix).

### CODE REVIEW

**Comment by willprice** [comment]
I'm rejecting this on the grounds that you can just open the URI as a stream using url:

```python
urllib.urlretrieve(file_name, 'file.mp4')
player = OMXPlayer('file.mp4')
player.play()
```

**Comment by tiangolo** [comment]
That would then disallow playing a live stream. It would have to be downloaded first to be played.

**Comment by willprice** [comment]
If you're happy to update your pull request to bring it up to date with the latest changes then I'll review and merge it in.

**Comment by tiangolo** [comment]
Cool! I'll do it. 

Should I do a rebase of my changes on top of the current `develop`? 
Should I merge the current `develop` in my branch? 
Or should I create a new PR with these same changes?

(Just to know how do you want to keep the "git history").

**Comment by willprice** [comment]
Hi Sebastian,

A rebase would be preferred, thanks for offering to bring this up to date :+1:

**Comment by tiangolo** [comment]
Done! :tada: 

I don't know what should I do with the `codecov/patch` "failing" test... (or if I should do anything).

**Comment by willprice** [comment]
Can we get a test to check if we check for the file to exist when we use a normal file path, and that we don't check when we're using a URI?

I'm also unsure about leaving the `filename` as its currently named, as now we're more generic, perhaps we should rename to `uri`, but that's not great either since it can either be a file path, or a URI. `uri_or_path` is fairly awful too. Any ideas?

**Comment by tiangolo** [comment]
OK, I agree. I'll work on it this week (right now I have to go).

**Comment by tiangolo** [comment]
I updated the repo tests. 

Nevertheless, I would like you to check the changes and see if they look good to you, as I'm not an expert with those specific testing tools.

---

What do you think of using `source` instead of `filename`?

If you approve (or chose a different variable name), I'll update all the references to use that.

**Comment by willprice** [comment]
Hi @tiangolo,

Thanks for you contributions.

`source` is an excellent idea! We need to make it very clear in the entry point that for most people this will be a video (on that note, I really need to fix up the RTD docs to be more friendly to newcomers).

It's quite late here, but I'll try and review your changes in detail tomorrow.

Thanks

**Comment by tiangolo** [comment]
I just refactored `player.py` to use `source` instead of `file_name` or `file_path`.

I added methods for `player.get_source` but left the `player.get_filename` there too for backward compatibility. Also the same with `player._load_source` and `player._load_file`.

I also updated a bit the docstrings to make it more clear that a "source" can be a file path or a URL.

Also, I updated the `README.md` with these live streaming options. And I added a `file_path_or_url` variable to both examples to make it more clear that it works with a file_path or a URL and that the syntax and usage is the same.

File: tests/test_omxplayer.py [hunk 1]
```diff
@@ -116,6 +117,11 @@ def test_checks_media_file_exists_before_launching_player(self, *args):
             self.patch_and_run_omxplayer()
             ospath.isfile.assert_called_once_with(self.TEST_FILE_NAME)
 
+    def test_not_checks_url_before_launching_player(self, *args):
```
Comment (willprice) [inline tests/test_omxplayer.py]: I think a better name is `test_player_doesnt_check_source_path_exists_for_a_url`

File: README.md [hunk 2]
```diff
@@ -53,7 +74,7 @@ player.pause()
 player.quit()
 ```
 
-Playing a file from a URL (courtesy of @jappe999)
+Playing a file from a URL by downloading the network file (courtesy of @jappe999)
```
Comment (willprice) [inline README.md]: I'm not sure about keeping this example any more now that we can pass URLs to omxplayer. Is there an instance where this would be preferred over instantiating the player with the URL instead?

File: omxplayer/player.py [hunk 3]
```diff
@@ -71,16 +72,20 @@ def __init__(self, filename,
...
         if self._process:
             self.quit()
 
-        self._process = self._setup_omxplayer_process(filename)
+        self._process = self._setup_omxplayer_process(source)
         self._connection = self._setup_dbus_connection(self._Connection, self._bus_address_finder)
 
-    def _run_omxplayer(self, filename, devnull):
+    # For backward compatibility
+    def _load_file(self, source):
```
Comment (tiangolo) [inline omxplayer/player.py]: OK, removing it...

File: omxplayer/player.py [hunk 4]
```diff
@@ -503,12 +509,21 @@ def quit(self):
...
 
     @_check_player_is_active
+    def get_source(self):
+        """
+        Returns:
+            str: source currently playing
+        """
+        return self._source
+
+    # For backward compatibility
```
Comment (tiangolo) [inline omxplayer/player.py]: Yep, much better idea.

File: omxplayer/player.py [hunk 5]
```diff
@@ -503,12 +509,21 @@ def quit(self):
...
+    @_check_player_is_active
     def get_filename(self):
         """
         Returns:
-            str: filename currently playing
+            str: source currently playing, for backward compatibility
```
Comment (tiangolo) [inline omxplayer/player.py]: I tried a couple options, and built the docs entering `./docs` and running `make dirhtml` to check that it looked OK and that the link worked. But I'm not that proficient with Sphinx docs, please check that it's looks how supposed to.

File: tests/test_omxplayer.py (hunk 1 above)
Comment (tiangolo) [inline tests/test_omxplayer.py]: Done :+1:

File: README.md (hunk 2 above)
Comment (tiangolo) [inline README.md]: I don't see a clear need for that example either.

Maybe the corner case that someone has a slow connection and wants to download the file completely before starting to play it. But still, I think that shouldn't be a problem that omxplayer-wrapper should "solve" and document.

I'm removing it too.

Comment (willprice) [inline README.md]: Cool, I'll add a note on the bottom of the README for suggested usage patterns like this one, but I don't think they should clutter the main usage example.

File: omxplayer/player.py (hunk 5 above)
Comment (willprice) [inline omxplayer/player.py]: Nice one, just checked out your pull request, all looks good to me too
